### ISSUE TITLE
immediate help in tensorflow

### ISSUE THREAD

**Issue by nikhilt23** [nikhilt23/tensorflow#1]
I am using tensorflow for 1D data. I want to apply CNN to it but it is showing errors. Could anyone help??

x= tf.placeholder(tf.float32, [None,9])
x= tf.reshape(x,shape=[100,9,1])
y_ = tf.placeholder(tf.float32, [None, 12])

#MODEL BEING BUILT

def weight_variable(shape):
  initial = tf.truncated_normal(shape, stddev=0.1)
  return tf.Variable(initial)

def bias_variable(shape):
  initial = tf.constant(0.1, shape=shape)
  return tf.Variable(initial)

def conv2d(x, W):
  return tf.nn.conv1d(x, W, stride=1, padding='SAME')

def max_pool_2x2(x):
  return tf.nn.max_pool(x, ksize=[1, 1, 2, 1],strides=[1, 1, 2, 1], padding='SAME')

def next_batch(num, dat, labels):
    '''
    Return a total of `num` random samples and labels. 
    '''
    idx = np.arange(0 , len(dat))
    np.random.shuffle(idx)
    idx = idx[:num]
    dat_shuffle = [dat[ i] for i in idx]
    labels_shuffle = [labels[ i] for i in idx]
    return np.asarray(dat_shuffle), np.asarray(labels_shuffle)

w_conv1 = weight_variable([3,1,5])
b_conv1= bias_variable([5])
w_conv2= weight_variable([2,5,8])
b_conv2= bias_variable([8])
w_fc1 = weight_variable([3*8,100])
b_fc1 = bias_variable([100])
w_fc2 = weight_variable([100,12])
b_fc2= bias_variable([12])

(X,y)= (raw_data[0],raw_data[1])
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=2)
X_train= np.array(X_train)
X_test= np.array(X_test)

X_train=X_train.reshape(X_train.shape[0],9,1)
X_test= X_test.reshape(X_test.shape[0],9,1)
Y_train = np_utils.to_categorical(y_train)
Y_test = np_utils.to_categorical(y_test)
#print(Y_train)
keep_prob = tf.placeholder(tf.float32)
def model(data):
	# ASSEMBLY OF NN
	#data= tf.reshape(data, shape=[-1,200,200,1])
	h_conv1 = tf.nn.relu(conv2d(data, w_conv1)+ b_conv1)
	h_pool1 = max_pool_2x2(h_conv1)
	h_conv2 = tf.nn.relu(conv2d(h_pool1, w_conv2)+ b_conv2)
	h_pool2 = max_pool_2x2(h_conv2)
	h_pool2_flat = tf.reshape(h_pool2, [-1,3*8])
	h_fc1 = tf.nn.relu(tf.matmul(h_pool2_flat,w_fc1) + b_fc1)
	h_fc1_drop= tf.nn.dropout(h_fc1, keep_prob)
	return tf.nn.softmax(tf.matmul(h_fc1_drop,w_fc2) + b_fc2)

y= model(x)
cross_entropy = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits = y, labels=y_))
optimizer = tf.train.GradientDescentOptimizer(0.05).minimize(cross_entropy)
correct_prediction = tf.equal(tf.argmax(y,1), tf.argmax(y_,1))
accuracy = tf.reduce_mean(tf.cast(correct_prediction,tf.float32))
sess.run(tf.global_variables_initializer())
saver = tf.train.Saver()

with tf.Session() as sess:
  sess.run(tf.global_variables_initializer())
  for i in range(20):
      for j in range(100):
        batch = next_batch(100, X_train, Y_train)
        optimizer.run(feed_dict={x:batch[0], y_: batch[1], keep_prob : 0.5})
      if i%1 == 0:
          train_accuracy = accuracy.eval(feed_dict={x:batch[0], y_: batch[1], keep_prob: 1.0})
          print("step %d, Training accuracy %g" %( i, train_accuracy))
  saver.save(sess, "C:/Users/Nikhil/Desktop/intern 2017/intern/model1",global_step=1000)  
print("test accuracy %g"%accuracy.eval(feed_dict={x: X_test, y_: Y_test, keep_prob: 1.0}))
